tests/scalers/temporal/temporal_test.go

```diff
@@ -0,0 +1,265 @@
+//go:build e2e
+// +build e2e
+
+package temporal_test
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/joho/godotenv"
+	. "github.com/kedacore/keda/v2/tests/helper"
+	"github.com/stretchr/testify/assert"
+	"k8s.io/client-go/kubernetes"
+)
+
+var _ = godotenv.Load("../../.env")
+
+const (
+	testName = "temporal-test"
+)
+
+var (
+	testNamespace = fmt.Sprintf("%s-ns", testName)
+
+	TemporalDeploymentName = fmt.Sprintf("temporal-%s", testName)
+
+	scaledObjectName = fmt.Sprintf("%s-so", testName)
+	deploymentName   = fmt.Sprintf("%s", testName)
+)
+
+type templateData struct {
+	DeploymentName         string
+	TestNamespace          string
+	TemporalDeploymentName string
+	ScaledObjectName       string
+}
+
+const (
+	temporalServiceTemplate = `
+apiVersion: v1
+kind: Service
+metadata:
+  name: {{.TemporalDeploymentName}}
+  namespace: {{.TestNamespace}}
+spec:
+  type: ClusterIP
+  ports:
+    - port: 7233
+      protocol: TCP
+      targetPort: 7233
+  selector:
+    app: temporal
+`
+
+	temporalDeploymentTemplate = `
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{.TemporalDeploymentName}}
+  namespace: {{.TestNamespace}}
+  labels:
+    app: temporal
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: temporal
+  template:
+    metadata:
+      labels:
+        app: temporal
+    spec:
+      containers:
+        - name: ubuntu
+          image: ubuntu:latest
```

Comment:
Please use the `temporalio/admintools` image. No need to download and install Temporal each time.